### PR TITLE
Implement initial version of product list

### PR DESCRIPTION
**Pull request by marcoskwkm**
#### What problem is this solving?

This implements a first version of the product list that will be used in IO Checkout. It currently lacks a proper layout but contains some minimum functionalities that can be tested, for instance, changing the quantity of each product or removing an item from the list.

#### How should this be manually tested?

[This workspace](https://productlist--vtexgame1.myvtex.com/cart) can be used to interact with the component.

#### Checklist/Reminders

- [ ] Updated `README.md`. (Will do in a future PR)
- [x] Updated `CHANGELOG.md`.
- [ ] Linked this PR to a Clubhouse story (if applicable).
- [x] Updated/created tests (important for bug fixes).
- [ ] Deleted the workspace after merging this PR (if applicable).

#### Screenshots or example usage
Screen Shot 2019-08-09 at 10 43 04 AM

#### Type of changes

✔️ | Type of Change
---|---
_ | Bug fix 
✔️ | New feature 
_ | Breaking change 
_ | Technical improvements

### CODE REVIEW

File: react/ProductList.tsx [hunk 1]
```diff
@@ -1,45 +1,31 @@
...
+const ProductList: FunctionComponent<Props> = ({
+  currency,
+  items,
+  onQuantityChange,
+  onRemove,
+}) => (
+  <div>
+    {items.map((item: any, index: number) => (
+      <ListItem
+        key={index}
```
Comment (klzns) [inline react/ProductList.tsx:21]: That's ok for now, but this will lead to bugs in the future

File: react/components/ListItem.tsx [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  <div className="flex bb b--muted-4 pv5 ph5">
+    <div className="mr6">
+      <a href={item.detailUrl}>
+        <img alt={item.name} src={item.imageUrl} width="96" height="96" />
+      </a>
+    </div>
+    <div>
+      <div className="lh-copy pb3">
+        <div>{item.additionalInfo.brandName}</div>
+        <a className="blue no-underline" href={item.detailUrl}>
```
Comment (klzns) [inline react/components/ListItem.tsx]:
```suggestion
        <Link className="c-link no-underline" to={item.detailUrl}>
```

File: react/components/ListItem.tsx [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+          }
+        >
+          <FormattedPrice
+            currency={currency}
+            value={item.listPrice * item.quantity}
+          />
+        </div>
+      </div>
+      <div>
+        <a className="gray no-underline" href="#" onClick={onRemove}>
```
Comment (klzns) [inline react/components/ListItem.tsx]:
```suggestion
        <a className="c-muted-1 no-underline" href="#" onClick={onRemove}>
```

File: react/components/QuantitySelector.tsx [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+    return (
+      <Dropdown
+        options={dropdownOptions}
+        size="small"
+        value={value}
+        onChange={(event: any) => handleChange(event.target.value)}
+        placeholder=""
+      />
+    )
+  } else {
```
Comment (klzns) [inline react/components/QuantitySelector.tsx:81]: remove else, is not necessary

**Review by klzns — DISMISSED** [review]

File: react/__tests__/ProductList.test.tsx [hunk 5]
```diff
@@ -0,0 +1,99 @@
+import React from 'react'
+import { render, fireEvent } from '@vtex/test-tools/react'
+
+import ProductList from '../ProductList'
+
+describe('Product List', () => {
+  const mockItems: Item[] = [
```
Comment (omninando) [inline react/__tests__/ProductList.test.tsx]: You should extract this code to a `__MOCKS__/mockItems`, so it's easier to read tests.

File: react/ProductList.tsx (hunk 1 above)
Comment (marcoskwkm) [reply react/ProductList.tsx:21]: What would be the proper way of assigning the `key` prop?

File: react/components/QuantitySelector.tsx (hunk 4 above)
Comment (marcoskwkm) [reply react/components/QuantitySelector.tsx:81]: I added the optional `else` because it keeps the code symmetric, which I believe improves readability.

**Review by omninando — CHANGES_REQUESTED** [review]

**Review by omninando — APPROVED** [review]

**Pull request merged**